### PR TITLE
Binary Tree

### PR DESCRIPTION
**Pull request by Hinal-Srivastava**
[#19](https://github.com/codesankalp/dsalgo/issues/19). A Transversing Binary Tree in Python is implemented.

### CODE REVIEW

**Comment by ravi5175** [comment]
@Hinal-Srivastava please provide more details like unittest or explaining more about your code

**Comment by codesankalp** [comment]
@Hinal-Srivastava please do these changes:
- follow our commit template.
- read our contributing guidelines
- read about how to link a issue with the pr (pr guidelines)
- add tests for your work
- attach screenshot here.

Thanks for contributing.

**Comment by codesankalp** [comment]
@Hinal-Srivastava are you still working on this?

**Comment by Hinal-Srivastava** [comment]
> @Hinal-Srivastava are you still working on this?

Yes. I'll update it by tonight

**Comment by Hinal-Srivastava** [comment]
> @Hinal-Srivastava please do these changes:
> 
> * follow our commit template.
> * read our contributing guidelines
> * read about how to link a issue with the pr (pr guidelines)
> * add tests for your work
> * attach screenshot here.
> 
> Thanks for contributing.

I didn't find any commit template

**Comment by Hinal-Srivastava** [comment]
image

File: dsalgo/binary_tree.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    return tree
+
+tree = binary_search_tree()    
+
+n = int(input('Number of Elements: '))
+for i in range(n):
+    value=int(input('Enter Element: '))
+    tree.insert(value)
+s = int(input('Search Element: '))
+print(tree.search(s))
```
Comment (17sushmita) [inline dsalgo/binary_tree.py:59]: Don't use driver code here. Make a file for unittesting and test the operations you've implemented in the tree. You can check `tests` directory in the repo for more context.

**Review by 17sushmita — CHANGES_REQUESTED** [review]
Read the documentation README.rst. We're developing a python library so you need not to test your code in the module. Make a separate test file in tests folder and implement unittesting.

**Comment by codesankalp** [comment]
@Hinal-Srivastava are you still working on this?

File: dsalgo/binary_tree.py (hunk 1 above)
Comment (codesankalp) [inline dsalgo/binary_tree.py:59]: add a new line at the end of line and follow pep8 guidelines or read our contributing.md

**Comment by devkapilbansal** [comment]
@Hinal-Srivastava please made the necessary changes as requested

**Comment by Aryamanz29** [comment]
Closing this PR, Since contributor is inactive
Thanks

**Pull request closed without merging**